Bookstore: connect to API

KIsmoilov opened the pull request:
**Bookstore: connect to API**

The following features have been implemented:

- Used fetch and axios with createAsyncThunk for making API calls.
- Fetched book list from Bookstore API on Books page load.
- Added an Action Creator, that returns a function.
- Refactored add book and remove book features to persist your changes in the server.
- Add book and remove book functions work in the same way after the refactor.

KIsmoilov committed:
Connect React-Redux app to an existing API to create and remove books in a remote server

muneebulrehman commented on src/components/Book.js:
```diff
@@ -1,25 +1,47 @@
+    title, author, category, item_id,
+  } = book;
+
+  useEffect(() => {
```
Though you did a great job with the application, however, `useEffect` hook you are using is causing an infinite loop of dispatching the action. Have a look at [this article](https://stackoverflow.com/questions/53120972/how-to-call-loading-function-with-react-useeffect-only-once). If you have any difficulty figuring it out, kindly leave me a message, but don't forget to tag me.

KIsmoilov replied:
@muneebulrehman Thanks for your review. If I run it once, the page is not refreshed after the remove button click. It would be great if you have time for a zoom call to explain it to me.

muneebulrehman reviewed (changes requested):
Hi @KIsmoilov , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

KIsmoilov committed:
Add new action creator

KIsmoilov committed:
Update the useEffect function and action creator functions

KIsmoilov committed:
Fix the linter errors

KIsmoilov commented:
**Bookstore: connect to API**

The following features have been implemented:

- Updated the useEffect function.
- Modified the actions creator functions.

mukhammadrizooff commented on src/redux/books/books.js:
```diff
@@ -1,54 +1,67 @@
+/* eslint-disable camelcase */
```
- [ ] [OPTIONAL] Good job so far! Please fix linter errors without this kind of comment `/* eslint-disable ... */` because this type of comment is used in very few cases, and in your case, you can solve such errors yourself, and I think if implement my suggestion that this will increase your work experience and skills 😃

mukhammadrizooff reviewed (approved):
Hi @KIsmoilov, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media3.giphy.com/media/sR9nilEznDoSlga667/giphy.gif"/>

## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
